scripts/mk_repo_file.py

```diff
@@ -191,8 +204,11 @@ def make_repo_file(url: str, ref: str,
 
             add_to_cache(p)
 
-            if callback is not None:
-                callback(data)
+            with cb_lock:
+                if callback is not None:
+                    callback(data)
+
+    pool.map(process_item, data.items())
```

Comment:
I think `pool.map` only returns a generator. It has to be evaluated explicitly using, e.g. `list`.

```suggestion
    list(pool.map(process_item, data.items()))
```

Reply:
No, the lazy version is `Pool.imap`.